Jasmine/Common/Models/CharacterGrid.swift

```diff
@@ -0,0 +1,83 @@
+struct CharacterGrid {
+    /// The actual grid.
+    private var grid: [[String?]]
```

Comment:
But in your ADT seems like it can have an array of varyingly sized array, so it might not live up to what you wanted it to be. Some rows might be shorter